File: src/main/java/banane/io/pdb/model/Action.java
package banane.io.pdb.model;

import javax.validation.constraints.NotNull;
import java.util.Optional;

public enum Action {

    MINE("MINE"),
    LOGGING("LOGGING"),
    CREATE_BASE("CREATE_BASE");

    private String name;

    private Action(String name) {
        this.name = name;
    }

    public String getName() {
        return this.name;
    }

    public static Optional<Action> parse(@NotNull String rawAction) {
        for (Action action : Action.values()) {
            if (action.name.equalsIgnoreCase(rawAction)) {
                return Optional.of(action);
            }
        }
        return Optional.empty();
    }

}

File: src/main/kotlin/banane/io/pdb/model/Action.kt
package banane.io.pdb.model

import java.util.*

enum class Action(val nameOfAction: String) {
    MINE("MINE"), LOGGING("LOGGING"), CREATE_BASE("CREATE_BASE"), VISIT_BASE("VISIT_BASE");

    companion object {
        fun parse(rawAction: String?): Optional<Action> {
            for (action in values()) {
                if (action.nameOfAction.equals(rawAction, ignoreCase = true)) {
                    return Optional.of(action)
                }
            }
            return Optional.empty()
        }
    }

}